trickster/model.py

```diff
@@ -251,6 +307,7 @@ class Route(BaseModel):
     responses: list[Response] = Field(default_factory=list, description='Possible responses of the route')
     response_selector: ResponseSelector = Field(
         default=ResponseSelector.RANDOM, description='Strategy for response selection')
+    auth: Union[Auth.get_subclasses()] | None = Field(default=None, discriminator='method')  # type: ignore
```

Comment:
This is nice and all, but if it doesn't work with mypy, I would personally rather explicitly fill all valid options `Union[TokenAuth, XYZAuth,]` and let the mypy be happy. But wait for others what they thinking of this...

Reply:
I recommended this to Erik as a solution. I see your point but the goal was also that Route doesn't have to have the responsibility of knowing what are the possible options. One solution might be declaring this type just bellow the Auth class like this:

`auth_types = cast(Auth | None, Union[Auth.get_subclasses()] | None)`

Or something like this. It should make mypy happy, pydantic happy and keep the responsibility of listing auth types outside.

Reply:
Or we can make the method return the typing type and not tuple. But that would require renaming it and making it very clear, what it's for 

I would prefer this solution and rather disable mypy here.